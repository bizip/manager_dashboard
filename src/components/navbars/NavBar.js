import React, { useEffect, useState } from 'react';
import { HiMenu } from 'react-icons/hi';
import { IoIosArrowDown } from 'react-icons/io';
import { useAuthState } from 'react-firebase-hooks/auth';
import { Link, useNavigate } from 'react-router-dom';
import {
  query, collection, getDocs, where,
} from 'firebase/firestore';
// import { onAuthStateChanged } from 'firebase/auth';

import { toast } from 'react-toastify';
import { auth, db } from '../auth/firebase';
import Dropdown from '../subNavBars/Dropdown';

const NavBar = () => {
  const [user, loading] = useAuthState(auth);
  const [name, setName] = useState('');
  const defaultPic = 'https://images.unsplash.com/photo-1438761681033-6461ffad8d80?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Nnx8cHJvZmlsZSUyMHBpY3R1cmV8ZW58MHx8MHx8&auto=format&fit=crop&w=500&q=60';
  const [profile, setProfile] = useState(defaultPic);

  // const [isLoggedIn, setIsLoggedIn] = useState(false);
  const [isDropDown, setIsDropDown] = useState(false);

  const toggleDropDown = () => {
    setIsDropDown((prev) => !prev);
  };
  const navigate = useNavigate();

  const fetchUserName = async () => {
    try {
      const q = query(collection(db, 'users'), where('uid', '==', user?.uid));
      const doc = await getDocs(q);
      const data = doc.docs[0].data();
      setName(data.name);
    } catch (err) {
      toast.error('An error occured while fetching user data', {
        position: 'top-right',
        autoClose: 5000,
        hideProgressBar: false,
        closeOnClick: true,
        pauseOnHover: true,
        draggable: true,
        progress: undefined,
        theme: 'light',
      });
    }
  };
  /* eslint-disable consistent-return */
  useEffect(() => {
    // if (loading) return;
    // On the above line we can add loader
    if (!user) return navigate('/');
    fetchUserName();
    user.providerData.forEach((profile) => {
      if (profile.photoURL) {
        setProfile(profile.photoURL);
      }
    });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [user, loading]);

  return (
    <div>
      <nav className="nav_container">
        <div className="nav__left">
          <h2>architect</h2>
          <HiMenu className="nav__hamberger" />
        </div>

        <div className="nav__right">
          {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
          <Link className="nav__avatar" onClick={toggleDropDown}>
            <img
              src={profile}
              className="avatar__profile"
              alt="Avatar"
            />

            <IoIosArrowDown className="avatar__dropdown" />
          </Link>

          <div className="avatar__details">
            <p className="details_header">{userName}</p>
            <p className="details_header--sub">{ user?.email }</p>
          </div>

        </div>
      </nav>
      <Dropdown isDropDown={isDropDown} />
    </div>
  );
};

export default NavBar;
