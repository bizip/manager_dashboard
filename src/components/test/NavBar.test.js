import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import NavBar from '../navbars/NavBar';

describe('Test for rendering dashboard page', () => {
  it('It should render dashboard page', () => {
    const navbar = renderer
      .create(<Router><NavBar /></Router>)
      .toJSON();
    expect(navbar).toMatchSnapshot();
  });
});
