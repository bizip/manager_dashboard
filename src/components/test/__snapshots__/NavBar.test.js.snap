// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for rendering dashboard page It should render dashboard page 1`] = `
<nav
  className="nav_container"
>
  <div
    className="nav__left"
  >
    <h2>
      architect
    </h2>
    <svg
      className="nav__hamberger"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="nav__right"
  >
    <div
      className="nav__avatar"
    >
      <img
        alt="Avatar"
        className="avatar__profile"
        src="https://images.unsplash.com/photo-1438761681033-6461ffad8d80?i xlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8Nnx8cHJvZmlsZSUyMHBpY3R1cmV8ZW 58MHx8MHx8&auto=format&fit=crop&w=500&q=60"
      />
      <svg
        className="avatar__dropdown"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
        />
      </svg>
    </div>
    <div
      className="avatar__details"
    >
      <p
        className="details_header"
      >
        Aliana Mclourd
      </p>
      <p
        className="details_header--sub"
      >
        VP people manager
      </p>
    </div>
  </div>
</nav>
`;
